test/integration/xds_integration_test.cc

```diff
@@ -36,5 +40,49 @@ TEST_P(XdsIntegrationTest, RouterRequestAndResponseWithBodyNoBuffer) {
   testRouterRequestAndResponseWithBody(1024, 512, false);
 }
 
+typedef HttpProtocolIntegrationTest LdsIntegrationTest;
+
+INSTANTIATE_TEST_SUITE_P(Protocols, LdsIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                             {Http::CodecClient::Type::HTTP1}, {FakeHttpConnection::Type::HTTP1})),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+// Sample test making sure our config framework correctly reloads listeners.
+TEST_P(LdsIntegrationTest, ReloadConfig) {
+  use_lds_ = true;
+  autonomous_upstream_ = true;
+  initialize();
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+  const std::string lds_filename =
+      config_helper_.bootstrap().dynamic_resources().lds_config().path();
+
+  // HTTP 1.0 is disabled by default.
+  std::string response;
+  sendRawHttpAndWaitForResponse(lookupPort("http"), "GET / HTTP/1.0\r\n\r\n", &response, true);
+  EXPECT_TRUE(response.find("HTTP/1.1 426 Upgrade Required\r\n") == 0);
+
+  // Turn up HTTP/1.0 proxying.
+  config_helper_.allowFurtherEdits();
+  config_helper_.addConfigModifier(
+      [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm) {
+        hcm.mutable_http_protocol_options()->set_accept_http_10(true);
+        hcm.mutable_http_protocol_options()->set_default_host_for_http_10("default.com");
+      });
+  config_helper_.applyConfigModifiers();
+
+  // Create an LDS response with the new config, and reload config.
+  envoy::api::v2::DiscoveryResponse lds = config_helper_.createLdsResponse("1");
+  std::string file = TestEnvironment::writeStringToFileForTest(
+      "new_lds_file", MessageUtil::getJsonStringFromMessage(lds));
+  TestUtility::renameFile(file, lds_filename);
+
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 2);
```

Comment:
Is listener_manager.lds.update_success == 1 after initialize()? If so, can we add an EXPECT there?